src/substitute.rs

```diff
@@ -180,9 +199,69 @@ fn substitute_line(
     Ok(Complete(output))
 }
 
+#[derive(Debug)]
+struct Pair {
+    open: String,
+    close: String,
+}
+
+#[derive(Debug)]
+struct ListSyntax {
+    separator: String,
+    pair: Option<Pair>,
+}
+
+fn parse_list_syntax(s: &str) -> anyhow::Result<ListSyntax> {
```

Comment:
It's not a particularly pretty implementation. I focused more on just getting it correct at first by doing some TDD development. Will see if I can come up with some more test cases maybe, and then I will look into refactoring this a bit